process/transaction/shardProcess.go

```diff
@@ -655,7 +655,7 @@ func (txProc *txProcessor) processUserTx(
 	returnCode := vmcommon.Ok
 	switch txType {
 	case process.MoveBalance:
-		err = txProc.processMoveBalance(userTx, userTx.SndAddr, userTx.RcvAddr, txProc.getUserTxCost(userTx, txHash, txType))
+		err = txProc.processMoveBalance(userTx, txProc.getUserTxCost(userTx, txHash, dstShardTxType), dstShardTxType)
```

Comment:
So also processMoveBalance and getUserTxCost will use dstShardTxType here?

Reply:
yes. the cost is according to that type - if it is moveBalance on destination shard - than we should take only moveBalance from that. otherwise we should take all.

Reply:
ok